dev/pll/device.c

```diff
@@ -183,3 +222,81 @@ void pll_calibrate_vco(pll_dev_t *pll, pll_freq_cal_t *cal, pll_freq_cfg_t *cfg)
         vco_get_fll_band_status_reg_if(p_vco, &p_vco_cal->band, &p_vco_cal->fine);
     }
 }
+
+void pll_set_loss_lock_interrupt_state(pll_dev_t *pll, bool enable)
+{
+    uint32_t reg_val;
+    reg_val = reg_read(pll->base + DDR_MVP_PLL_CORE_STATUS_INT_EN__ADR);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_MVP_PLL_CORE_STATUS_INT_EN_LOSS_OF_LOCK_INT_EN, enable);
+    reg_write(pll->base + DDR_MVP_PLL_CORE_STATUS_INT_EN__ADR, reg_val);
+}
+
+void pll_set_lock_interrupt_state(pll_dev_t *pll, bool enable)
+{
+    uint32_t reg_val;
+    reg_val = reg_read(pll->base + DDR_MVP_PLL_CORE_STATUS_INT_EN__ADR);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_MVP_PLL_CORE_STATUS_INT_EN_CORE_LOCKED_INT_EN, enable);
+    reg_write(pll->base + DDR_MVP_PLL_CORE_STATUS_INT_EN__ADR, reg_val);
+}
+
+void pll_set_init_lock_interrupt_state(pll_dev_t *pll, bool enable)
+{
+    uint32_t reg_val;
+    reg_val = reg_read(pll->base + DDR_MVP_PLL_CORE_STATUS_INT_EN__ADR);
+    reg_val = UPDATE_REG_FIELD(reg_val, DDR_MVP_PLL_CORE_STATUS_INT_EN_INITIAL_SWITCH_DONE_INT_EN, enable);
+    reg_write(pll->base + DDR_MVP_PLL_CORE_STATUS_INT_EN__ADR, reg_val);
+}
+
+void pll_get_current_vco(pll_dev_t *pll, uint8_t *vco_id)
+{
+    *vco_id = pll->p_vco_current->vco_id;
+}
+
+void pll_get_current_freq(pll_dev_t *pll, uint8_t *freq_id)
+{
+    *freq_id = pll->p_vco_current->freq_id;
+}
```

Comment:
I didn't want to add next vco / frequency because of they're complication with potentially being NULL. Additionally, we didn't need those functions anymore so I just removed them.